src/main/java/com/redhat/labs/lodestar/service/GitService.java

```diff
@@ -0,0 +1,337 @@
+package com.redhat.labs.lodestar.service;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.json.bind.Jsonb;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.GenericType;
+import javax.ws.rs.core.Response;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.labs.lodestar.model.Artifact;
+import com.redhat.labs.lodestar.model.Engagement;
+import com.redhat.labs.lodestar.model.gitlab.Branch;
+import com.redhat.labs.lodestar.model.gitlab.File;
+import com.redhat.labs.lodestar.model.gitlab.Project;
+import com.redhat.labs.lodestar.model.gitlab.ProjectTree;
+import com.redhat.labs.lodestar.model.gitlab.ProjectTreeNode;
+import com.redhat.labs.lodestar.model.pagination.PagedResults;
+import com.redhat.labs.lodestar.rest.client.GitlabRestClient;
+
+@ApplicationScoped
+public class GitService {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(GitService.class);
+
+	private static final String ARTIFACT_FILE = "artifacts.json";
+	private static final String ENGAGEMENT_FILE = "engagement.json";
+
+	@ConfigProperty(name = "group.parent.id")
+	Integer groupParentId;
+
+	@ConfigProperty(name = "default.branch")
+	String defaultBranch;
+
+	@ConfigProperty(name = "default.commit.message")
+	String defaultCommitMessage;
+
+	@ConfigProperty(name = "default.author.name")
+	String defaultAuthorName;
+
+	@ConfigProperty(name = "default.author.email")
+	String defaultAuthorEmail;
+
+	@ConfigProperty(name = "default.page.size")
+	Integer pageSize;
+
+	@Inject
+	@RestClient
+	GitlabRestClient gitlabRestClient;
+
+	@Inject
+	Jsonb jsonb;
+
+	/**
+	 * Returns the name of the default branch for the given project ID. If not
+	 * found, the configured value is returned.
+	 * 
+	 * @param projectId
+	 * @return
+	 */
+	String findDefaultBranch(Integer projectId) {
+
+		List<Branch> branches = gitlabRestClient.findProjectBranches(projectId);
+		Optional<Branch> branch = branches.stream().filter(Branch::getIsDefault).findAny();
+
+		return branch.isPresent() ? branch.get().getName() : defaultBranch;
+
+	}
```

Comment:
The default branch is returned in the get project by uuid call

Reply:
actually may not in the search. However, we should assume a default branch name

Reply:
using configured default branch.  still not sure what happens when we have a combination of `main` and `master`.